.github/workflows/teslight-update-packaging-tool-and-tup-file-build.yml

```diff
@@ -0,0 +1,78 @@
+name: TesLight Update Packaging Tool and TUP file Build
+on:
+  push:
+    branches: ["main", "releases/**", "feat/DevOps_Auto_build_for_Update_Packages"]
+    tags: ["v*", "v**", "v.*", "v.**" ]
+  pull_request:
+    branches: ["main", "releases/**"]
+  workflow_dispatch:
+  workflow_call:
+jobs:
+  build_update_tool_for_different_os:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    # prevent stuck jobs
+    timeout-minutes: 20
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Update Tool Build
+        uses: ./.github/actions/update-tool-build
+
+  generate_tup_file:
+    needs: [build_update_tool_for_different_os]
+    runs-on: ubuntu-latest
+    # prevent stuck jobs
+    timeout-minutes: 20
```

Comment:
This is only for generating the TUP? This should take a few seconds max😋.

Reply:
As mentioned above, Windows and macOS runners are really slow. Linux build is done in 24 seconds, windows and macOS can take 1.5 minutes or 12 minutes, you never know...